pkg/src/index.js

```diff
@@ -436,6 +443,58 @@ export async function publint({ pkgDir, vfs, level, strict, _packedFiles }) {
     }
   }
 
+  /**
+   * @param {Record<string, string> | string} repositoryField
+   */
+  async function checkRepositoryField(repositoryField) {
+    if (typeof repositoryField === 'string') {
+      if (!isRepositoryUrl(repositoryField)) {
+        messages.push({
+          code: 'INVALID_REPOSITORY_VALUE',
+          args: {},
+          path: ['repository'],
+          type: 'warning',
+        });
+      }
+
+      return;
+    } else if (typeof repositoryField === 'object') {
+      if (repositoryField.url && !isRepositoryUrl(repositoryField.url)) {
+        messages.push({
+          code: 'INVALID_REPOSITORY_VALUE',
+          args: {},
+          path: ['repository', 'url'],
+          type: 'warning',
+        });
+      }
+
+      if (repositoryField.directory) {
+        const altRootPath = vfs.pathJoin(rootPkgPath, repositoryField.directory);
+        const isAltRootExist = await vfs.isPathExist(altRootPath);
```

Comment:
Is this correct? If we have a package at `<root>/packages/my-pkg`, then the `directory` should be `packages/my-pkg`, but `rootPkgPath` would be `<root>/packages/my-pkg/package.json`.

I think we need to crawl upwards and find a `.git` folder, and then try to match the directory, but it's a bit complex and I think we can skip checking this for now.

Reply:
Aah, it seems like I misunderstand this field, my bad. This field is exclusively used for monorepo.

Does `publint` currently supports monorepo? I think we can implement this check *if* we are able to deal with monorepo.

Reply:
Not really for now, it only checks per-package each time, so it doesn't understand about monorepos. It's also something we can't check in the browser as the tarball only contains the package itself.

Reply:
Fair, let's remove this check for now.